share/picongpu/tests/PusherScaling/lib/python/test/Pusher/MainTest.py

```diff
@@ -0,0 +1,285 @@
+
+        # all radii in the series
+        radius = (
+            np.sqrt(x_momentum**2 + y_momentum**2)
+            * self.params["unit_mass"]
+            * self.params["unit_speed"]
+            / (abs(self.charge[0]) * self.params["unit_charge"] * self.B)
+            / self.params["cell_depth"]
+            / self.params["unit_length"]
+        )
+
+        R_c = radius[0]  # original radius
+
+        # half_step
+        steps_per_turn = 0.825e-12 / (self.params["dt"] * self.params["unit_time"])
+        halfStepPhase = np.pi / steps_per_turn
+
+        # coordinate transformation to reposition the origin at the center of
+        # the circle
+        x_poss = x_poss + (R_c * np.sin(halfStepPhase))
+        y_poss = y_poss - R_c + int(R_c) + (R_c - R_c * np.cos(halfStepPhase))
+
+        x_offSet = x_offSet - 32
+        y_offSet = y_offSet - 5 - int(R_c)
+
+        self.period = period
+
+        # in-cell position
+        self.x_poss = x_poss
+        self.y_poss = y_poss
+
+        # cells
+        self.x_offSet = x_offSet
+        self.y_offSet = y_offSet
+
+        # momenta of the series
+        self.x_momentum = x_momentum
+        self.y_momentum = y_momentum
+
+        self.R_c = R_c
+
+    def get_total_phasediff(self):
+        """Calculates the total phase difference accumulated during the
+        simulation.
+
+        Returns:
+        total_phasediff (float): Total phase difference
+        timestep (int): Timestep used in the series
+
+        """
+        x = self.x_poss + self.x_offSet  # x coordinates
+        y = self.y_poss + self.y_offSet  # y coordinates
+
+        res_cos = x / np.sqrt(x**2 + y**2)
+        theta = np.arccos(res_cos)
+
+        timestep = self.params["dt"] * self.params["unit_time"]
+
+        total_phasediff = 0
+        for counter in range(0, len(theta) - 1, 1):
+            total_phasediff += abs(theta[counter + 1] - theta[counter])
+
+        return total_phasediff, timestep
+
+
+def main():
+    """Main function. Creates a compare object and calls its functions to
+    compare PIConGPU's phase difference scaling with the expectation.
+
+    Returns:
+    compare_result (int): 0 if the simulated scaling can be regarded as
+                          equivalent to the expectation;
+                          Non-zero if they differ.
+
+    """
+    B = 50
+    # array for total phase differences
+    phaseDifferences = np.zeros(len(sys.argv) - 1)
+    timesteps = np.zeros(len(sys.argv) - 1)
+
+    for j in range(1, len(sys.argv)):
+        path = sys.argv[j]
+        phase_class = Comparison(path, B)
+        phase_class.get_params()
+
+        # extract all relevant data
+        phase_class.read_series()
+
+        phaseDifferences[j - 1], timesteps[j - 1] = phase_class.get_total_phasediff()
+
+    quotients = np.zeros(len(phaseDifferences) - 1)
+    for j in range(len(phaseDifferences) - 1):
+        quotients[j] = phaseDifferences[j + 1] / phaseDifferences[j]
```

Comment:
```suggestion
        quotients = phaseDifferences[1:] / phaseDifferences[:-1]
```
to replace these 3 lines.